state/state.go

```diff
@@ -1729,7 +1729,16 @@ func (st *State) AddRelation(eps ...Endpoint) (r *Relation, err error) {
 		return ops, nil
 	}
 	if err = st.run(buildTxn); err == nil {
-		return &Relation{st, *doc}, nil
+		return &Relation{st, *doc}, err
+	} else if errors.IsAlreadyExists(err) {
+		// If the relation already exists, we'll return it
```

Comment:
I checked with Menno and he was ok with this, saying the pattern is used elsewhere too. What this does save doing it here is a facade rpc call.